app/src/main/java/org/wikipedia/dataclient/okhttp/OkHttpWebViewClient.kt

```diff
@@ -47,6 +47,9 @@ abstract class OkHttpWebViewClient : WebViewClient() {
         var response: WebResourceResponse
         try {
             val rsp = request(request)
+            if (rsp.networkResponse != null && rsp.cacheResponse == null && request.url.encodedPath?.contains(RestService.PAGE_HTML_ENDPOINT) == true) {
```

Comment:
I'm not sure I see how this is working. Why are we calling `pageFetchStart()` after the request has already returned?

We should call `pageFetchStart()` before the `request()` is made (1 line above), and then `pageFetchEnd()` after it's finished, i.e. right here. That's all that is needed. We shouldn't need to keep track of the URL inside the schema, or need to call `pageFetchEnd` when the page is finished _painting_ (that's not what we're measuring).

Reply:
Got it and done!